shimmingtoolbox/prepare_fieldmap.py

```diff
@@ -97,28 +108,54 @@ def prepare_fieldmap(list_nii_phase, echo_times, mag, unwrapper='prelude', mask=
 
         # Calculate the echo time difference
         echo_time_diff = echo_times[1] - echo_times[0]  # [s]
+        # Run the unwrapper
+        phasediff_unwrapped = unwrap_phase(nii_phasediff, unwrapper=unwrapper, mag=mag, mask=mask,
+                                           fname_save_mask=fname_save_mask)
 
-    else:
-        # TODO: More echoes
-        # TODO: Add method once multiple methods are implemented
-        raise NotImplementedError(f"This number of phase input is not supported: {len(phase)}.")
-
-    # Run the unwrapper
-    phasediff_unwrapped = unwrap_phase(nii_phasediff, unwrapper=unwrapper, mag=mag, mask=mask,
-                                       fname_save_mask=fname_save_mask)
+        # If it's 4d (i.e. there are timepoints)
+        if len(phasediff_unwrapped.shape) == 4:
+            phasediff_unwrapped = correct_2pi_offset(phasediff_unwrapped, mag, mask, VALIDITY_THRESHOLD)
 
-    # If it's 4d (i.e. there are timepoints)
-    if len(phasediff_unwrapped.shape) == 4:
-        phasediff_unwrapped = correct_2pi_offset(phasediff_unwrapped, mag, mask, VALIDITY_THRESHOLD)
-
-    # Divide by echo time
-    fieldmap_rad = phasediff_unwrapped / echo_time_diff  # [rad / s]
-    fieldmap_hz = fieldmap_rad / (2 * math.pi)  # [Hz]
+        # Divide by echo time
+        fieldmap_rad = phasediff_unwrapped / echo_time_diff  # [rad / s]
+        fieldmap_hz = fieldmap_rad / (2 * math.pi)  # [Hz]
 
+    else:
+        # Calculates field map based on multi echo phases by running the prelude unwrapper for each phase individually.
+        unwrapped = [unwrap_phase(list_nii_phase[echo_number], unwrapper=unwrapper, mag=mag, mask=mask,
+                                  fname_save_mask=fname_save_mask) for echo_number in range(len(list_nii_phase))]
+        n_echoes = len(list_nii_phase)  # Number of Echoes
+        n_y = unwrapped[0].shape[0]  # Number of voxels along Y direction (rows) in each volume.
+        n_x = unwrapped[0].shape[1]  # Number of voxels along X direction (columns) in each volume.
+        n_z = unwrapped[0].shape[2]  # Number of slices along Z direction in each volume.
+        list_unwrapped_nii = []
+        for echo in range(n_echoes):
+            list_unwrapped_nii.append(unwrapped[echo])
```

Comment:
You can avoid this expansive for loop by using `unwrapped` which already contains all the data you need.
You can write over `unwrapped` (which should probably be renamed to `nii_unwrapped`)

```
nii_unwrappred = np.moveaxis(np.stack(nii_unwrapped, axis=0), 0, 3)
```